app/src/test/java/org/xbmc/kore/ui/sections/remote/SharedUrlConverterTest.java

```diff
@@ -0,0 +1,197 @@
+package org.xbmc.kore.ui.sections.remote;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.*;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = 23)
+public class SharedUrlConverterTest {
+    private static final String YT_PLUGIN_PREFIX = "plugin://plugin.video.youtube/play/?video_id=";
+    private static final String VM_PLUGIN_PREFIX = "plugin://plugin.video.vimeo/play/?video_id=";
+    private static final String YT_ID = "vOHNyS8KPYY";
+    private static final String VM_ID = "12345678";
+    private static final String TWITCH_ID = "arteezy";
+    private static final String GARBAGE_BEFORE = "asdf sdf sdf ";
+    private static final String GARBAGE_AFTER = " asdf sdf sdf";
+
+    private String twitchUrl(String id) {
+        return "plugin://plugin.video.twitch/playLive/" + id + "/";
+    }
+
+    @Test
+    public void twitch_actual_shared_text() {
```

Comment:
Just a habit. I don't see tests as methods, they're more like specs or documentation to me so they tend to have long names. SnakeCase gets really weird to read when there's a dozen words especially when there's the article 'a' somewhere.